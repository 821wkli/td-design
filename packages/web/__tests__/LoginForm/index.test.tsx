import React from 'react';
import { shallow, mount } from 'enzyme';
import LoginForm from '../../components/login-form';

describe('测试LoginForm组件', () => {
  it('1. 成功渲染', () => {
    const wrapper = shallow(<LoginForm placeholder="请输入手机号码" onSubmit={() => {}} phone />);
    expect(wrapper).toMatchSnapshot();
  });

  it('2. 测试placeholder属性', () => {
    const wrapper = mount(<LoginForm placeholder="请输入手机号码" onSubmit={() => {}} phone />);
    expect(wrapper.find('input.phone').prop('placeholder')).toEqual('请输入手机号码');
  });

  it('3. 测试onSubmit方法是否被传入', () => {
    const onSubmit = jest.fn();
    const wrapper = mount(<LoginForm placeholder="请输入手机号码" onSubmit={onSubmit} phone />);
    expect(wrapper.props().onSubmit).toBeDefined();
  });

  it('4. 测试用户名登录', () => {
    const onSubmit = jest.fn();
    const wrapper = mount(<LoginForm placeholder="请输入用户名" onSubmit={onSubmit} />);
    expect(
      wrapper
        .find('input')
        .at(0)
        .hasClass('userName'),
    ).toBe(true);
  });

  it('5. 测试手机号登录', () => {
    const onSubmit = jest.fn();
    const wrapper = mount(<LoginForm placeholder="请输入手机号码" onSubmit={onSubmit} phone />);
    expect(
      wrapper
        .find('input')
        .at(0)
        .hasClass('phone'),
    ).toBe(true);
  });

});
