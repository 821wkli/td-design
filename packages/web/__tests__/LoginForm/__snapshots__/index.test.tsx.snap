// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`测试LoginForm组件 1. 成功渲染 1`] = `
<LoginForm
  form={
    Object {
      "getFieldDecorator": [Function],
      "getFieldError": [Function],
      "getFieldInstance": [Function],
      "getFieldProps": [Function],
      "getFieldValue": [Function],
      "getFieldsError": [Function],
      "getFieldsValue": [Function],
      "isFieldTouched": [Function],
      "isFieldValidating": [Function],
      "isFieldsTouched": [Function],
      "isFieldsValidating": [Function],
      "isSubmitting": [Function],
      "resetFields": [Function],
      "setFields": [Function],
      "setFieldsInitialValue": [Function],
      "setFieldsValue": [Function],
      "submit": [Function],
      "validateFields": [Function],
      "validateFieldsAndScroll": [Function],
    }
  }
  onSubmit={[Function]}
  phone={true}
  placeholder="请输入手机号码"
/>
`;
